clang/test/CodeGenSYCL/disable_loop_pipelining.cpp

```diff
@@ -29,8 +29,7 @@ int main() {
   return 0;
 }
 
-// CHECK: define dso_local spir_kernel void @"{{.*}}test_kernel1"() #0 !kernel_arg_buffer_location ![[NUM4:[0-9]+]] !disable_loop_pipelining ![[NUM5:[0-9]+]]
-// CHECK: define dso_local spir_kernel void @"{{.*}}test_kernel2"() #0 !kernel_arg_buffer_location ![[NUM4]]
-// CHECK: define dso_local spir_kernel void @"{{.*}}test_kernel3"() #0 !kernel_arg_buffer_location ![[NUM4]] !disable_loop_pipelining ![[NUM5]]
-// CHECK: ![[NUM4]] = !{}
+// CHECK: define dso_local spir_kernel void @"{{.*}}test_kernel1"() #0 {{.*}} !disable_loop_pipelining ![[NUM5:[0-9]+]]
+// CHECK: define dso_local spir_kernel void @"{{.*}}test_kernel2"() #0 {{.*}}
+// CHECK: define dso_local spir_kernel void @"{{.*}}test_kernel3"() #0 {{.*}} !disable_loop_pipelining ![[NUM5]]
 // CHECK: ![[NUM5]] = !{i32 1}
```

Comment:
This seems unrelated to the patch. Is there any reason you are modifying the lit test?

Reply:
The position of !kernel_arg_buffer_location has changed because I inserted !noalias in the string, due to which this test is failing.

Reply:
I see. Thanks for the explanation.

Reply:
It has to be modified anyway, since `noalias` metadata starts to be generated. The way the test is modified in this patch is IMO preferable, since disable_loop_pipelining.cpp shan't check neither buffer_location nor noalias metadata.

Reply:
I agree. Thanks @MrSidims for the explanation.